shell/src/context_listener.rs

```diff
@@ -188,6 +191,7 @@ fn listen_protocol_events(
     apply_block_run: &AtomicBool,
     event_server: &mut IpcEvtServer,
     event_server_accept_timeout: Duration,
+    actions_file_storage: &mut Option<ActionFileStorage>,
```

Comment:
arent `actions_file_storage` kind of the same thing ? they seems to store actions on demand. Maybe ActionFileStorage and `ContextActionStorage` should be implementing some sort of abstraction and throught configuration it should be possible to choose one of those ?

Reply:
Its kinda the same thing but its much faster to fetch from ActionFile and RockDB use a lot memory so it might squee memory statistics , but we can build an abstraction over it.

Reply:
`actions_file_storage` is also deleted when the node restarts